docs/src/cookbooks/serialization.md

```diff
@@ -0,0 +1,177 @@
+}
+
+// Parameter
+case class GCDSerializableModuleParameter(width: Int) extends SerializableModuleParameter
+
+// Module
+class GCDSerializableModule(val parameter: GCDSerializableModuleParameter)
+    extends Module
+    with SerializableModule[GCDSerializableModuleParameter] {
+  val io = IO(new Bundle {
+    val a = Input(UInt(parameter.width.W))
+    val b = Input(UInt(parameter.width.W))
+    val e = Input(Bool())
+    val z = Output(UInt(parameter.width.W))
+  })
+  val x = Reg(UInt(parameter.width.W))
+  val y = Reg(UInt(parameter.width.W))
+  val z = Reg(UInt(parameter.width.W))
+  val e = Reg(Bool())
+  when(e) {
+    x := io.a
+    y := io.b
+    z := 0.U
+  }
+  when(x =/= y) {
+    when(x > y) {
+      x := x - y
+    }.otherwise {
+      y := y - x
+    }
+  }.otherwise {
+    z := x
+  }
+  io.z := z
+}
+```
+using `write` function in `upickle`:
+```scala
+val j = upickle.default.write(
+  SerializableModuleGenerator(
+    classOf[GCDSerializableModule],
+    GCDSerializableModuleParameter(32)
+  )
+)
+println(j)
+```
+
+read from json string and elaborate the Module:
+```scala
+println(chisel3.stage.ChiselStage.emitVerilog(
+  upickle.default.read[SerializableModuleGenerator[GCDSerializableModule, GCDSerializableModuleParameter]](
+    ujson.read(j)
+  ).module()
+))
+```
+```verilog
+module GCDSerializableModule(
+  input         clock,
+  input         reset,
+  input  [31:0] io_a,
+  input  [31:0] io_b,
+  input         io_e,
+  output [31:0] io_z
+);
+`ifdef RANDOMIZE_REG_INIT
+  reg [31:0] _RAND_0;
+  reg [31:0] _RAND_1;
+  reg [31:0] _RAND_2;
+`endif
+  reg [31:0] x;
+  reg [31:0] y;
+  reg [31:0] z;
+  wire [31:0] _x_T_1 = x - y;
+  wire [31:0] _y_T_1 = y - x;
+  assign io_z = z;
+  always @(posedge clock) begin
+    if (x != y) begin
+      if (x > y) begin
+        x <= _x_T_1;
+      end
+    end
+    if (x != y) begin
+      if (!(x > y)) begin
+        y <= _y_T_1;
+      end
+    end
+    if (!(x != y)) begin
+      z <= x;
+    end
+  end
+// Register and memory initialization
+`ifdef RANDOMIZE_GARBAGE_ASSIGN
+`define RANDOMIZE
+`endif
+`ifdef RANDOMIZE_INVALID_ASSIGN
+`define RANDOMIZE
+`endif
+`ifdef RANDOMIZE_REG_INIT
+`define RANDOMIZE
+`endif
+`ifdef RANDOMIZE_MEM_INIT
+`define RANDOMIZE
+`endif
+`ifndef RANDOM
+`define RANDOM $random
+`endif
+`ifdef RANDOMIZE_MEM_INIT
+  integer initvar;
+`endif
+`ifndef SYNTHESIS
+`ifdef FIRRTL_BEFORE_INITIAL
+`FIRRTL_BEFORE_INITIAL
+`endif
+initial begin
+  `ifdef RANDOMIZE
+    `ifdef INIT_RANDOM
+      `INIT_RANDOM
+    `endif
+    `ifndef VERILATOR
+      `ifdef RANDOMIZE_DELAY
+        #`RANDOMIZE_DELAY begin end
+      `else
+        #0.002 begin end
+      `endif
+    `endif
+`ifdef RANDOMIZE_REG_INIT
+  _RAND_0 = {1{`RANDOM}};
+  x = _RAND_0[31:0];
+  _RAND_1 = {1{`RANDOM}};
+  y = _RAND_1[31:0];
+  _RAND_2 = {1{`RANDOM}};
+  z = _RAND_2[31:0];
+`endif
+  `endif
+end
+`ifdef FIRRTL_AFTER_INITIAL
+`FIRRTL_AFTER_INITIAL
+`endif
+`endif
+endmodule
+```
```

Comment:
```suggestion
You can then read from json string and elaborate the Module:
```scala mdoc:silent
chisel3.stage.ChiselStage.emitVerilog(
  upickle.default.read[SerializableModuleGenerator[GCDSerializableModule, GCDSerializableModuleParameter]](
    ujson.read(j)
  ).module()
)
```
````
Assuming it will still compile (and just can't be run), you can use `mdoc:silent` here. Also I don't see value in showing the emitted Verilog, especially since it will get stale and we'll forget to update it. I think it's fine to just show the API.

Reply:
I change `mdoc:silent` into `mdoc:compile-only`, see: https://scalameta.org/mdoc/docs/modifiers.html#compile-only